src/parser/jobs/gnb/modules/AmmoCombo.tsx

```diff
@@ -0,0 +1,250 @@
+import {Trans} from '@lingui/react'
+import {ActionLink} from 'components/ui/DbLink'
+import {RotationTable} from 'components/ui/RotationTable'
+import ACTIONS from 'data/ACTIONS'
+import {Event, Events} from 'event'
+import _ from 'lodash'
+import {Analyser} from 'parser/core/Analyser'
+import {filter, oneOf} from 'parser/core/filter'
+import {dependency} from 'parser/core/Injectable'
+import Checklist, {Requirement, Rule} from 'parser/core/modules/Checklist'
+import {Data} from 'parser/core/modules/Data'
+import {Timeline} from 'parser/core/modules/Timeline'
+import React from 'react'
+
+// Expected within a standard gnashing.
+const EXPECTED_USES = {
+	JUGULAR_RIP: 1,
+	SAVAGE_CLAW: 1,
+	ABDOMEN_TEAR: 1,
+	WICKED_TALON: 1,
+	EYE_GOUGE: 1,
+	TOTAL_SKILLS_PER_COMBO: 6,
+}
+
+class GnashingComboState {
+	startTime: number
+	endTime?: number
+	rotation: Array<Events['action']>
+	isProper: boolean = false
+
+	constructor(start: number) {
+		this.startTime = start
+		this.rotation = []
+	}
+
+	public get totalHits(): number { return this.rotation.length }
+
+	public get actualJug() : number {
+		return this.rotation.filter(event => event.action === ACTIONS.JUGULAR_RIP.id).length
+	}
+
+	public get actualSavage(): number {
+		return this.rotation.filter(event => event.action === ACTIONS.SAVAGE_CLAW.id).length
+	}
+
+	public get actualTear(): number {
+		return this.rotation.filter(event => event.action === ACTIONS.ABDOMEN_TEAR.id).length
+	}
+
+	public get actualTalon(): number {
+		return this.rotation.filter(event => event.action === ACTIONS.WICKED_TALON.id).length
+	}
+
+	public get actualEye(): number {
+		return this.rotation.filter(event => event.action === ACTIONS.EYE_GOUGE.id).length
+	}
```

Comment:
This is on the edge of the type of getters we try to avoid using.  From a raw performance standpoint, it would be a lot better to just have `jugUsed: number = 0`, `savageUsed: number = 0`, etc. buckets (or even just booleans, since it's a 0 or 1 thing for these skills) and then just track when the skills get used, rather than filtering the rotation array 5 times.  Also, RotationTable winds up calling the getters multiple times per row when you pass the getter to the target property, so it compounds the problem.  

Alternatively, since Gnashing Fang combos are strictly linear and the player doesn't have a choice to go back or use in special orders, could we just drop all of the individual skill target columns in favor of showing the total X/6 skills used and the actual rotation used, and then remove these getters altogether?

Reply:
I'm not opposed to the 6/6 option, as even though you can add extenders to this combo, they aren't counted. So it'll just be the end user that has to see what's missing instead of us explicitly saying it